homeassistant/components/point/binary_sensor.py

```diff
@@ -121,7 +121,7 @@ def is_on(self):
         if self.device_class == BinarySensorDeviceClass.CONNECTIVITY:
             # connectivity is the other way around.
             return not self._is_on
-        return self._is_on
+        return bool(self._is_on)
```

Comment:
Looking at the code, `_is_on` can be `None`, `True` or `False`.
Previously this was actually a bug, as the only allowed values for `is_on` are `True` and `False`. However, for 2022.2 this has been changed to allow for `None` too.

So the solution here, is to type the method, as no conversion is needed.

```py
    @property
    def is_on(self) -> bool | None:
        """Return the state of the binary sensor."""
        if self.device_class == BinarySensorDeviceClass.CONNECTIVITY:
            # connectivity is the other way around.
            return not self._is_on
        return self._is_on
```

(ps: don't forget to add `from __future__ import annotations` at the top of the file)

Reply:
If I don't do this in my dev environment all the binary_sensors shows up an `Unknown` (where it actually should show `Normal` or `Off`.

Reply:
> all the binary_sensors shows up an Unknown

That would mean: `self._is_on` = `None`, in those cases Unknown is the right state? As the integration doesn't know the state? (Otherwise it would have been True or False, right?)

Reply:
I guess changing the updates to `False` is better than `None` like this:

```python
    async def _update_callback(self):
        """Update the value of the sensor."""
        if not self.is_updated:
            return
        if self._events[0] in self.device.ongoing_events:
            self._is_on = True
        else:
            self._is_on = False
        self.async_write_ha_state()

    @callback
    def _webhook_event(self, data, webhook):
        """Process new event from the webhook."""
        if self.device.webhook != webhook:
            return
        _type = data.get("event", {}).get("type")
        _device_id = data.get("event", {}).get("device_id")
        if _type not in self._events or _device_id != self.device.device_id:
            return
        _LOGGER.debug("Received webhook: %s", _type)
        if _type == self._events[0]:
            self._is_on = True
        if _type == self._events[1]:
            self._is_on = False
        self.async_write_ha_state()
```

Reply:
Is this better:

```python
    async def _update_callback(self):
        """Update the value of the sensor."""
        if not self.is_updated:
            return
        self._is_on = self._events[0] in self.device.ongoing_events
        self.async_write_ha_state()
```

Reply:
uh yeah possibily.

Additionally, instead of using `self._is_on`, you could also directly use `self._attr_is_on`, which removes the need for using a local variable and allows you to remove the `is_on` property method completely.

Reply:
I did change to use `self._attr_is_on` and it have been running for a while without any issues.